src/Open3D/GUI/Checkbox.h

```diff
@@ -0,0 +1,58 @@
+// ----------------------------------------------------------------------------
+// -                        Open3D: www.open3d.org                            -
+// ----------------------------------------------------------------------------
+// The MIT License (MIT)
+//
+// Copyright (c) 2018 www.open3d.org
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+// ----------------------------------------------------------------------------
+
+#pragma once
+
+#include "Widget.h"
+
+#include <functional>
+
+namespace open3d {
+namespace gui {
+
+class Checkbox : public Widget {
+public:
+    explicit Checkbox(const char* name);
+    ~Checkbox();
+
+    bool IsChecked() const;
+    void SetChecked(bool checked);
+
+    Size CalcPreferredSize(const Theme& theme) const override;
+
+    DrawResult Draw(const DrawContext& context) override;
+
+    /// Specifies a callback function which will be called when the box
+    /// changes checked state as a result of user action.
+    void SetOnChecked(std::function<void(bool)> onChecked);
```

Comment:
I've never like "toggle" for checkbox, since a checkbox isn't a button (despite the fact that, codewise, it probably is one). We have a "SetChecked" to set the value, so I'd expect a "SetOnChecked" to call the function when it is either checked or not checked (especially since there is a bool argument). If you really feel strongly about "SetOnChecked", another option is "SetValue" and "SetOnValueChanged", which would be consistent with other widgets.